nixops/backends/__init__.py

```diff
@@ -7,52 +7,19 @@
 import nixops.util
 import nixops.resources
 import nixops.ssh_util
-import xml.etree.ElementTree as ET
 
 
 class MachineDefinition(nixops.resources.ResourceDefinition):
     """Base class for NixOps machine definitions."""
 
-    def __init__(self, xml, config={}) -> None:
-        nixops.resources.ResourceDefinition.__init__(self, xml, config)
-        self.store_keys_on_machine = (
-            xml.find("attrs/attr[@name='storeKeysOnMachine']/bool").get("value")
-            == "true"
-        )
-        self.ssh_port = int(xml.find("attrs/attr[@name='targetPort']/int").get("value"))
```

Comment:
I couldn't remember if this would seek down in to the tree or if it was a "rooted" expression:

I wrote this XML:

```xml
<?xml version="1.0" encoding="UTF-8"?>

<foo>
  <attrs> 
    <attr name="targetPort"> 
      <int>1</int> 
    </attr> 
  </attrs>
</foo>
```

and this xpath expression: `attrs/attr[@name='targetPort']/int` and it didn't find it, but did when I removed the wrapping `foo`. In other words, this code didn't have the edge case I was worried about.

I guess that means the Backends are initialized with the `deployment` attrset as the config?

Reply:
> I guess that means the Backends are initialized with the deployment attrset as the config?

Yes, that's correct.